src/Desktop/UIAutomation/CustomObjects/Config.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Newtonsoft.Json;
+using System;
+using System.IO;
+
+namespace Axe.Windows.Desktop.UIAutomation.CustomObjects
+{
+    class Config
+    {
+        [JsonProperty("properties")]
+        public CustomProperty[] Properties { get; set; }
+
+        public static Config FromText(string text)
+        {
+            Config conf = JsonConvert.DeserializeObject<Config>(text);
+            conf.Validate();
+            // We made it here, so config must be valid.
+            return conf;
+        }
+
+        public static Config FromFile(string path) { return Config.FromText(File.ReadAllText(path)); }
+
+        private void Validate()
+        {
+            if (this.Properties == null || this.Properties.Length < 1) throw new ArgumentException("Empty or missing definition of custom properties.");
+            foreach (CustomProperty p in this.Properties) p.Validate();
```

Comment:
It would be preferable, to me anyway, if the for statement and the executed code were on separate lines. That makes it easier for me to separate the loop from the action taken within the loop. The same applies to most if conditions. The one place I like to have the condition and the execution on the same line is when doing parameter or returned data validation. Those statements are just boiler plate and having them on the same line means I know I can ignore them. New lines are a great tool for finding what is important and ignoring what isn't, or ignoring what can safely be assumed.